File: fixtures/scale/ruby/large/lib/check_sdbmhash.rb
require_relative 'sdbmhash_digest'
require_relative 'sdbmhash_digest_pair'
require_relative 'corpus_hash'

def check_sdbmhash
  corpus_hash.each do |text|
    first = sdbmhash_digest(text)
    second = sdbmhash_digest(text)
    return false if first != second || first.length != 8
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_sdbmhash.py
from cipherlab.ciphers.sdbmhash import sdbmhash_digest
from cipherlab.specs.sdbmhash_spec import sdbmhash_spec
from cipherlab.corpus.corpus_hash import corpus_hash


def check_sdbmhash():
    spec = sdbmhash_spec()
    for text in corpus_hash():
        first = sdbmhash_digest(text)
        second = sdbmhash_digest(text)
        if first != second or len(first) != 8:
            return False
    return spec["category"] == "hash"
